packages/get-starknet/src/wallet.ts

```diff
@@ -220,13 +239,107 @@ export class MetaMaskSnapWallet implements StarknetWindowObject {
+   */
+  off<Event extends keyof WalletEventHandlers>(event: Event, handleEvent: WalletEventHandlers[Event]): void {
+    if (event === WalletEvent.AccountsChanged) {
+      this.#accountChangeHandlers = this.#accountChangeHandlers.filter((handler) => handler !== handleEvent);
+    } else if (event === WalletEvent.NetworkChanged) {
+      this.#networkChangeHandlers = this.#networkChangeHandlers.filter((handler) => handler !== handleEvent);
+    } else {
+      throw new Error(`Unsupported event: ${String(event)}`);
+    }
+  }
+
+  /**
+   * Polling function for detecting changes with a maximum delay.
+   * The function balances between responsiveness (handling changes as quickly as possible)
+   * and efficiency (ensuring a controlled polling frequency).
+   *
+   * 1. Polling operation: Runs with a timeout to prevent infinite hangs.
+   * 2. Minimum delay: Introduces a small delay (e.g., 100ms) between iterations
+   * to avoid excessive resource usage while still scanning for updates promptly.
+   */
+  #pollingFunction = async (): Promise<void> => {
+    if (!this.#pollingController) {
+      return; // Abort if the polling controller is not initialized
+    }
+    const { signal } = this.#pollingController;
+
+    while (!signal.aborted) {
+      const previousNetwork = this.#chainId;
+      const previousAddress = this.#selectedAddress;
+
+      try {
+        // Perform the polling operation with a timeout.
+        // Ensures the operation completes within a maximum time frame to avoid infinite hangs.
+        // If `this.#init()` takes too long, the timeout will reject and continue to the next iteration.
+        await Promise.race([
+          // Fetch network, assign address and chainId for thread safe.
+          this.#init(),
+          new Promise((_, reject) =>
+            // Timeout after `MetaMaskSnapWallet.pollingTimeoutMs`.
+            setTimeout(() => reject(new Error('Polling timeout exceeded')), MetaMaskSnapWallet.pollingTimeoutMs),
+          ),
+        ]);
+
+        // Check for network change
+        if (previousNetwork !== this.#chainId) {
+          await Promise.allSettled(
+            this.#networkChangeHandlers.map(async (callback) =>
+              resolver(callback, this.#chainId, [this.#selectedAddress]),
+            ),
+          );
+        }
+
+        // Check for account change
+        if (previousAddress !== this.#selectedAddress) {
+          await Promise.allSettled(
+            this.#accountChangeHandlers.map(async (callback) => resolver(callback, [this.#selectedAddress])),
+          );
+        }
+      } catch (_error) {
+        // Not to throw an error, to avoid breaking the event subscription
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, MetaMaskSnapWallet.minimumPollingDelay));
+    }
+  };
+
+  #startPolling(): void {
+    this.#pollingController = new AbortController();
+    this.#pollingFunction().catch((_error) => {
+      // Do nothing
+    });
```

Comment:
i think we are safe to let the error not catch
as in the function itself, it has try catch on every scan
but if it has an error outside the while loop, then it is unexpected, we can throw that error
```suggestion
    this.#pollingFunction();
```